src/main/java/org/openelisglobal/patient/service/PatientServiceImpl.java

```diff
@@ -832,4 +844,45 @@ public Patient getByExternalId(String id) {
         return baseObjectDAO.getByExternalId(id);
     }
 
+
+    private Errors validatePatientInfo(PatientManagementInfo patientInfo) {
```

Comment:
Why dont you just add this validation function in the controller wherw it was before ??